pool_automation/scripts/inventory-init.py

```diff
@@ -0,0 +1,184 @@
+            elif type(d) in (int, float):
+                _help_kwargs['type'] = type(d)
+
+            _group.add_argument("--{}.{}".format(role, p), **_help_kwargs)
+
+    parser.add_argument("--logconfig", metavar="PATH", default=None,
+                        help=("Path to json-formatted logging configuration"
+                              " file, if not defined the one the basic"
+                              " one will be used"))
+
+    parser.add_argument("--show-defaults", action="store_true",
+                        help="Show defaults and exit")
+
+    return vars(parser.parse_args())
+
+
+def _dump_defaults(roles):
+    for role, params in roles.iteritems():
+        if 'defaults' in params:
+            print(role.upper())
+            print(yaml.safe_dump(params['defaults'], default_flow_style=False))
+
+
+def _dump_config(config_path, config_vars, default_vars):
+    """Dumps user specified config and commented out defaults"""
+
+    # TODO consider to use Ansible's 'to_nice_yaml' from
+    # ansible.plugins.filter.core.py BUT it's not a public API
+    with open(config_path, "w") as _f:
+        _f.write('---\n')
+        if config_vars:
+            yaml.safe_dump(config_vars, _f, default_flow_style=False)
+        _s = yaml.safe_dump(default_vars, default_flow_style=False)
+        _f.write(''.join(["\n# DEFAULTS\n\n"] + ["#{}".format(_l) for _l in _s.splitlines(True)]))
+
+
+def _specify_localhost(inventory_dir):
+    localhost_spec = {
+        'all': {
+            'hosts': {
+                'localhost': {
+                    'ansible_connection': 'local',
+                    'ansible_python_interpreter': '{{ ansible_playbook_python }}'
+                }
+            }
+        }
+    }
+
+    with open(os.path.join(inventory_dir, 'localhost.yml'), "w") as _f:
+        _f.write('---\n')
+        yaml.safe_dump(localhost_spec, _f, default_flow_style=False)
+
+
+def main():
+
+    _set_logging()
+
+    roles = _init_roles_defaults()
+
+    args = _parse_args(roles)
+
+    # output default api settings
+
+    # config logging
+    if args['logconfig'] is not None:
+        _set_logging(args['logconfig'])
+
+    logger.debug("Cmd line arguments: {}".format(args))
+
+    if args["show_defaults"]:
+        _dump_defaults(roles)
+        exit(0)
+
+    # create inventory dir hierarchy
+    group_vars_dir = "{}/group_vars/all".format(args['inventory-dir'])
+    if not os.path.isdir(group_vars_dir):
+        os.makedirs(group_vars_dir)
+
+    # dump configs
+    for role, params in roles.iteritems():
+        if 'defaults' not in params:
+            continue
+
+        # construct user specified config parameters
+        config = {}
+        for _p in params['defaults'].keys():
+            _arg_name = "{}.{}".format(role, _p)
+            if args.get(_arg_name):
+                config[_p] = args[_arg_name]
+
+        _dump_config("{}/{}_config.yml".format(group_vars_dir, role), config, params['defaults'])
```

Comment:
Having single config file per group instead of per role might be more manageable, at least now

Reply:
But it would be less readable since defaults will be accumulated for all roles.

Reply:
Done